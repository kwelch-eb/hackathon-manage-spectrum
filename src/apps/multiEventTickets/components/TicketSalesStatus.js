import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from 'eventbrite_design_system/icon/Icon';
import EBPropTypes from 'js-utils/prop-types';

export default class TicketSalesStatus extends PureComponent {
    static propTypes = {
        iconColor: PropTypes.string.isRequired,
        iconType: PropTypes.string.isRequired,
        displayStatus: EBPropTypes.translation.isRequired,
    };
    
    render() {
        let {iconColor, iconType, displayStatus} = this.props;
        let statusIconClass = classnames(
            'eds-show-up-mn',
            'eds-ticket-card-status-icon',
            'eds-l-mar-right-1',
        );
    
        return (<div className="eds-g-cell eds-l-mar-right-3">
            <Icon
                type={iconType}
                color={iconColor}
                size="xsmall"
                className={statusIconClass}
            /> 
            <span>{displayStatus}</span>
        </div>);
    }
}
