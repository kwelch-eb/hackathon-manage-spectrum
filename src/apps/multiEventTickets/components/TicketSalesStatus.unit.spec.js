import React from 'react';
import {mount} from 'enzyme';
import gettext from 'js-utils/gettext';

import TicketSalesStatus from './TicketSalesStatus';

describe('<TicketSalesStatus />', () => {
    it('should render', () => {
        let wrapper = mount(
            <TicketSalesStatus
                iconType="status-dot-chunky"
                iconColor="ui-green"
                displayStatus={gettext('On Sale')}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
