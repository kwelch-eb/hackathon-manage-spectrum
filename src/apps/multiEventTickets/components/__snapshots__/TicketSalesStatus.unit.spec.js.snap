// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TicketSalesStatus /> should render 1`] = `
<TicketSalesStatus
  displayStatus="On Sale"
  iconColor="ui-green"
  iconType="status-dot-chunky"
>
  <div
    className="eds-g-cell eds-l-mar-right-3"
  >
    <Icon
      className="eds-show-up-mn eds-ticket-card-status-icon eds-l-mar-right-1"
      color="ui-green"
      size="xsmall"
      type="status-dot-chunky"
    >
      <VectorImage
        color="ui-green"
        size="xsmall"
        svg={
          <VectorIcon
            type="status-dot-chunky"
          />
        }
        type="icon"
      >
        <i
          aria-hidden={true}
          className="eds-vector-image eds-icon--xsmall eds-vector-image--ui-green"
          data-spec="icon"
        >
          <VectorIcon
            type="status-dot-chunky"
          >
            <svg
              dangerouslySetInnerHTML={
                Object {
                  "__html": "
    <circle cx=\\"12\\" cy=\\"12\\" r=\\"3\\"/>
",
                }
              }
              id="eds-icon--alert-chunky_svg"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            />
          </VectorIcon>
        </i>
      </VectorImage>
    </Icon>
    <span>
      On Sale
    </span>
  </div>
</TicketSalesStatus>
`;
